tests/cross_fw/test_templates/template_test_weights_compression.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2025 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import math
+from abc import ABC
+from abc import abstractmethod
+from typing import List, TypeVar
+
+import numpy as np
+import pytest
+
+from nncf import CompressWeightsMode
+from nncf import SensitivityMetric
+from nncf.data.dataset import Dataset
+from nncf.quantization import compress_weights
+from nncf.quantization.algorithms.weight_compression.mixed_precision import MIXED_PRECISION_CRITERIA
+from nncf.tensor import TensorDataType
+
+TModel = TypeVar("TModel")
+TTensor = TypeVar("TTensor")
+
+NON_ZERO_ROW = [-4, 1, 2]
+ACTIVATION = [[NON_ZERO_ROW, [0, 0, 0], [0, 0, 0]]]
+MAX_VAR = 3.555555  # np.max(np.var(ACTIVATION, 1))
+MEAN_VAR = 1.555555  # np.mean(np.var(ACTIVATION, 1))
+MEAN_MAX = 2.333333  # np.mean(np.max(np.abs(ACTIVATION), 1))
+HESSIAN_TRACE = (16 + 1 + 4) * 2 / 9  # sum(i*i for i in NON_ZERO_ROW) * 2 / ACTIVATION.size
+MAX_BASELINE_SCORE = 1 / 1.1920928955078125e-07
+
+
+class TemplateWeightCompression(ABC):
```

Comment:
Nice! Actually, this should close this GFI https://github.com/openvinotoolkit/nncf/issues/3129